### PR TITLE
samples: Add example snippets for stale reads

### LINKED ISSUE #1002: Add examples for Snapshot Reads
Add example snippets for snapshot reads.

### PR DESCRIPTION
**Pull request by jlara310**
As part of feature work for snapshot reads, add examples for the documentation.

Fixes #1002  ☕️

### CODE REVIEW

**Comment by jlara310** [comment]
@yixiaoshen, please review.

File: samples/snippets/src/main/java/com/example/firestore/snippets/ManageDataSnippets.java [hunk 1]
```diff
@@ -468,6 +473,58 @@ public void updateDocumentIncrement() throws ExecutionException, InterruptedExce
...
+    Query query = cities.whereEqualTo("capital", true);
+
+    // In the transaction options, Set read time and and set to read-only
+    // Snapshot reads require read-only transactions
+    // As an example, set read time to the update time of the previous write operation
+    final Timestamp readTime = com.google.protobuf.Timestamp.newBuilder()
+        .setSeconds(writeResult.getUpdateTime().getSeconds())
+        .setNanos(writeResult.getUpdateTime().getNanos())
+        .build();
+    // final Timestamp readTime = writeResult.getUpdateTime();
```
Comment (yixiaoshen) [inline samples/snippets/src/main/java/com/example/firestore/snippets/ManageDataSnippets.java:502]: why do we have this commented line which returns the same timestamp as the above lines that constructs timestamp from the builder? is this for demonstration purpose?

**Review by yixiaoshen — APPROVED** [review]
LGTM

File: samples/snippets/src/main/java/com/example/firestore/snippets/ManageDataSnippets.java (hunk 1 above)
Comment (jlara310) [reply samples/snippets/src/main/java/com/example/firestore/snippets/ManageDataSnippets.java]: Left in that comment by mistake. Removed. Thanks!

**Review by tom-andersen — APPROVED** [review]

**Pull request merged**